File: Javascript/bio/src/map/Map.js
Class('bio.map.Map', {
	constructor: function Map() {
		this.cellSize = 10;
		this.colums = 100;
		this.rows = 100;

		this.cells = new bio.map.CellManager();
		this.elements = new bio.map.ElementManager();
	},

	properties: [ 'cellSize', 'columns', 'rows' ],


	tick: function() {
		this.elements.tick();
		//this.cells.tick();
	},

	reset: function() {
		this._configureChilds();
		this.cells.reset();
		return this;
	},

	_configureChilds: function() {
		this.cells.setCellSize(this.cellSize);
		this.cells.setColumns(this.columns);
		this.cells.setRows(this.rows);

		this.elements.setSize(this.cellSize * this.columns, this.cellSize * this.rows);
	},


	//
	// Element Manager methods
	//

	addElement: function(element) {
		this.elements.add(element, this.cells);
	},

	removeElement: function(element) {
		this.elements.remove(element);
	},

	updateLocation: function(element) {
		this.elements.updateLocation(element, this.cells);
	},

	getShorterDistance: function(element1, element2) {
		return this.elements.getShorterDistance(element1, element2);
	},

	getShorterAngle: function(element1, element2) {
		return this.elements.getShorterAngle(element1, element2);
	},


	//
	// Cell Manager methods
	//
/*
	getCellsAtElement: function(element) {
		return this.cells.getCellsAtElement(element);
	},
*/
	getRangeFromElement: function(element, radius) {
		return this.cells.getRangeFromElement(element, radius);
	}
});

File: Dart/life/specs/map/Map.dart
class Map {
	int cellSize = 10;
	int columns = 100;
	int rows = 100;

	final var cells = new CellManager();
	final var element = new ElementManager();


	//
	// Methods
	//

	tick() => elements.tick();

	reset() {
		cells.cellSize = cellSize;
		cells.columns = columns;
		cells.rows = rows;

		elements.setSize(cellSize * columns, cellSize * rows);
		cells.reset();
	}


	//
	// ElementManager Methods
	//

	addElement(Element element) =>
		lements.add(element, cells);
	removeElement(Element element) =>
		elements.remove(element);

	updateLocation(Element element) =>
		elements.updateLocation(element, cells);

	num shorterDistance(Element element1, Element element2) =>
		elements.shorterDistance(element1, element2);
	num shorterAngle(Element element1, Element element2) =>
		elements.shorterAngle(element1, element2);


	//
	// CellManager methods
	//

	getRangeFromElement(Element element, num radius) =>
		cells.getRangeFromElement(element, radius);
}